File: Comparative_Analysis_Exercise/JS/studentRegChecker.js
// Student Registration Status Checker in JS

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question("Enter student name: ", (studentName) => {
    readline.question("Enter number of registered units: ", (unitsInput) => {
        const units = unitsInput;
        
        // Determine registration status
        const status = units > 7 ? "Overload - Approval Required" : "Registration Accepted";
        
        // Print final summary
        console.log("\n========== REGISTRATION SUMMARY ==========");
        console.log(`Student Name: ${studentName}`);
        console.log(`Registered Units: ${units}`);
        console.log(`Status: ${status}`);
        console.log("==========================================\n");
        
        readline.close();
    });
});

File: Comparative_Analysis_Exercise/student_reg_checker.py
# Student Registration Status Checker in Python

stu_name = str(input("Enter your name: "))
units = int(input("Enter no. of units registered: "))

response = f"Student Name: {stu_name.capitalize()}\nUnits Registered: {units}\nStatus: {'Overload - Approval Required.' if units > 7 else 'Registration Accepted.'}"


if units > 7:
    print("-"*30)
    print(response)
    print("-"*30)
else:
    print("-"*30)
    print(response)
    print("-"*30)